qemu/tests/boot_from_virtiofs.py

```diff
@@ -0,0 +1,107 @@
+import os
+
+from avocado.utils import process
+from virttest import data_dir
+from virttest import env_process
+
+
+def create_kernel_initrd(test, params):
+    """
+    Create initramfs and kernel file with dracut modul
+    """
+    test.log.info("Creating initramfs and kernel file")
+    install_path = data_dir.get_data_dir()
+    kernel_version = process.getoutput(params.get("guest_ver_cmd", "uname -r"))
+    create_initramfs_cmd = (params["create_initramfs_cmd"] % install_path)
+    status, output = process.getstatusoutput(create_initramfs_cmd)
+    if status:
+        test.fail("Failed to create initramfs.")
+    test.log.info("initramfs is created in %s" % install_path)
+    initrd_path = install_path + "/initramfs-virtiofs.img"
+
+    # copy vmlinuz to virtiofs_root
+    process.system("cp /boot/vmlinuz-%s %s" % (kernel_version, install_path))
+    kernel_path = install_path + ("/vmlinuz-%s" % kernel_version)
+    params["kernel"] = kernel_path
+    params["initrd"] = initrd_path
+    return [kernel_path, initrd_path]
+
+
+def setup_basic_root_fs(test, params):
+    """
+    create basic root file system
+    """
+    test.log.info("Setting basic root file system")
+    virtiofs_root_path = data_dir.get_data_dir() + "/virtiofs_root"
+    install_file_system_cmd = (params["install_file_system_cmd"]
+                               % virtiofs_root_path)
+    status, output = process.getstatusoutput(install_file_system_cmd)
+    if status:
+        test.fail("Failed to install basic root file system."
+                  "Error message: %s" % output)
+    return virtiofs_root_path
+
+
+def change_fs_passwd(test, params, virtiofs_root_path):
+    """
+    change the password
+    """
+    test.log.info("Changing password of the virtiofs")
+    check_selinux_cmd = params["check_selinux_cmd"]
+    selinux_value = process.getoutput(check_selinux_cmd)
+    if selinux_value != "Disabled":
+        process.getoutput("setenforce 0")
```

Comment:
We should record the current value and then restore it when test finished if we are changing host environment.